pkg/handler/common_repositories.go

```diff
@@ -0,0 +1,74 @@
+package handler
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/dao"
+	ce "github.com/content-services/content-sources-backend/pkg/errors"
+	"github.com/labstack/echo/v4"
+)
+
+type CommonRepositoriesHandler struct {
+	RepositoryDao dao.RepositoryConfigDao
+}
+
+func RegisterCommonRepositoriesRoutes(
+	engine *echo.Group,
+	dao *dao.RepositoryConfigDao,
+) {
+	rph := CommonRepositoriesHandler{RepositoryDao: *dao}
+	engine.GET("/common_repositories/", rph.listCommonRepositories)
+}
+
+// ListCommonRepositories godoc
+// @Summary      List Common Repositories
+// @ID           listCommonRepositories
+// @Description  get common repositories
+// @Tags         common,repositories
+// @Accept       json
+// @Produce      json
+// @Success      200 {object} api.CommonRepositoriesCollectionResponse
+// @Router       /common_repositories/ [get]
+func (rh *CommonRepositoriesHandler) listCommonRepositories(c echo.Context) error {
+	jsonConfig, err := os.ReadFile("./pkg/config/common_repositories.json")
+
+	if err != nil {
+		return ce.NewErrorResponseFromError("listCommonRepositories could not read common_repositories.json", err)
+	}
+
+	configData := []api.CommonRepositoryResponse{}
+
+	err = json.Unmarshal([]byte(jsonConfig), &configData)
+	if err != nil {
+		return ce.NewErrorResponseFromError("listCommonRepositories could not read common_repositories.json", err)
+	}
+
+	for i := 0; i < len(configData); i++ {
+		err := rh.updateDataIfExists(c, &configData[i])
+
+		if err != nil {
+			return ce.NewErrorResponseFromError("listCommonRepositories could not Get RepositoryDao.List", err)
+		}
+	}
+
+	return c.JSON(200, setCollectionResponseMetadata(&api.CommonRepositoriesCollectionResponse{Data: configData}, c, int64(len(configData))))
+}
+
+func (rh *CommonRepositoriesHandler) updateDataIfExists(c echo.Context, repo *api.CommonRepositoryResponse) error {
+	_, orgID := getAccountIdOrgId(c)
+	// Go get the records for this URL
+	repos, _, err := rh.RepositoryDao.List(orgID, api.PaginationData{}, api.FilterData{Search: repo.URL})
+	if err != nil {
+		return ce.NewErrorResponseFromError("listCommonRepositories could not Get RepositoryDao.List", err)
+	}
+
+	// If the URL exists update the "existingName" field
+	if len(repos.Data) > 0 && repos.Data[0].Name != "" {
+		repo.ExistingName = repos.Data[0].Name
+		repo.UUID = repos.Data[0].UUID
+	}
+
+	return nil
```

Comment:
a much more efficient way to do this, would be to do 1 query for all urls, and the collate the lists together.  Obviously for 4 repos it doesn't matter too much, but you might at least leave a comment hinting at that solution

Reply:
I could leave a comment or write a separate db query. Which would you prefer here?